packages/commerce-sdk-react/src/hooks/query.test.tsx

```diff
@@ -0,0 +1,211 @@
+    //     hook: () => usePublicProductListsBySearchTerm(),
+    //     endpoint: '',
+    // },
+    // {
+    //     name: 'usePublicProductList',
+    //     hook: () => usePublicProductList(),
+    //     endpoint: '',
+    // },
+    // {
+    //     name: 'useProductListItem',
+    //     hook: () => useProductListItem(),
+    //     endpoint: '',
+    // },
+
+
+    {
+        name: 'useOrder',
+        hook: () => useOrder({orderNo: '123'}),
+        endpoint: '/orders/123',
+    },
+    {
+        name: 'usePaymentMethodsForOrder',
+        hook: () => usePaymentMethodsForOrder({orderNo: '123'}),
+        endpoint: '/orders/123/payment-methods',
+    },
+    {
+        name: 'useTaxesFromOrder',
+        hook: () => useTaxesFromOrder({orderNo: '123'}),
+        endpoint: '/orders/123/taxes',
+    },
+
+
+    {
+        name: 'useProducts',
+        hook: () => useProducts({ids: '123,456'}),
+        endpoint: '/products',
+    },
+    {
+        name: 'useProduct',
+        hook: () => useProduct({id: '123'}),
+        endpoint: '/products/123',
+    },
+    {
+        name: 'useCategories',
+        hook: () => useCategories({ids: '123,456'}),
+        endpoint: '/categories',
+    },
+    {
+        name: 'useCategory',
+        hook: () => useCategory({id: '123'}),
+        endpoint: '/categories/123',
+    },
+    {
+        name: 'usePromotions',
+        hook: () => usePromotions({ids: '123,456'}),
+        endpoint: '/promotions',
+    },
+    {
+        name: 'usePromotionsForCampaign',
+        hook: () => usePromotionsForCampaign({campaignId: '123'}),
+        endpoint: '/promotions/campaigns/123',
+    },
+
+    {
+        name: 'useProductSearch',
+        hook: () => useProductSearch({q: 'test'}),
+        endpoint: '/product-search',
+    },
+    {
+        name: 'useSearchSuggestions',
+        hook: () => useSearchSuggestions({q: 'test'}),
+        endpoint: '/search-suggestions',
+    },
+]
+
+test.each(QUERY_TESTS)('%j - 200 returns data', async ({hook, endpoint}) => {
+    const data = {test: true}
+    nock('http://localhost:8888')
+        .get((uri) => uri.includes(endpoint))
+        .reply(200, data)
+    // @ts-ignore
+    const {result, waitForNextUpdate} = renderHookWithProviders(hook)
+    expect(result.current.data).toBe(undefined)
+    expect(result.current.isLoading).toBe(true)
+
+    await waitForNextUpdate()
+
+    expect(result.current.isLoading).toBe(false)
+    expect(result.current.data).toEqual(data)
+})
+
+test.each(QUERY_TESTS)('%j - 400 returns error', async ({hook, endpoint}) => {
+    nock('http://localhost:8888')
+        .get((uri) => uri.includes(endpoint))
+        .reply(400)
+    // @ts-ignore
+    const {result, waitForNextUpdate} = renderHookWithProviders(hook)
+    expect(result.current.data).toBe(undefined)
+    expect(result.current.isLoading).toBe(true)
+
+    await waitForNextUpdate()
+
+    expect(result.current.isLoading).toBe(false)
+    expect(result.current.error).toBeTruthy()
```

Comment:
This line is the difference between the previous success tests with these error tests. But is it necessary to test it? 

I think we can assume that it is in the domain of react query and that the error handling will have been tested there already.

Reply:
This line test the fact that we are using isomorphic client correctly with the `throwOnResponse` flag.


But yeah, i don't have a strong opinion on this. Let me know again if you think we should remove this test.

Reply:
Oh I see, let's keep the test then.